### PR TITLE
Enforce proper usage of `py.typed` file

### LINKED ISSUE #24372: Update `verifywhl/verifysdist` to confirm presence of `py.typed` file
1. Verify `setup.py` arguments contain `include_package_data` set to `True`
2. Verify `setup.py` contains a `package_data` argument that includes at least one dictionary key that contains value `py.typed`.

### PR DESCRIPTION
**Pull request by scbedd**
Fixes #24372 

We had a discussion on a teams channel about a couple of our packages not properly having the `py.typed` file present. This PR extends the `verifySdist` tox environment to check for the proper configuration of said file.

### CODE REVIEW

**Comment by scbedd** [comment]
@weshaggard on Monday can you merge the commit from [this branch](https://github.com/Azure/azure-sdk-for-python/tree/add-missing-pytyped) into main?

(specifically asking Monday to avoid weekend pain 👍 )

Instead of having a long tail of cleanup to ensure that THIS PR doesn't break a bunch of builds, I just pulled the trigger on some combined auto/manual updates for the whole repo.

**Comment by weshaggard** [comment]
@scbedd just ping me monday to remind me but yes I can do it.

File: eng/tox/import_all.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def should_run_import_all(package_name):
 
     # get target package name from target package path
     pkg_dir = os.path.abspath(args.target_package)
-    package_name, namespace, _ = get_package_details(os.path.join(pkg_dir, 'setup.py'))
+    package_name, namespace, _, _ = get_package_details(os.path.join(pkg_dir, 'setup.py'))
```
Comment (mccoyp) [inline eng/tox/import_all.py:44]: Seems to be missing one return value:
```suggestion
    package_name, namespace, _, _, _ = get_package_details(os.path.join(pkg_dir, 'setup.py'))
```

Comment (scbedd) [reply eng/tox/import_all.py:44]: 100% agree. Thanks McCoy.

**Review by mccoyp — APPROVED** [review]

**Pull request merged**